File: services/word-cloud-analytic/src/main/scala/PathTranslator.scala
import java.util.Calendar

object PathTranslator {
    def getS3Path(offset: Int = 0): String = {
        val now = Calendar.getInstance()
        now.add(Calendar.DAY_OF_YEAR, -offset)
        val currentYear = now.get(Calendar.YEAR)
        val currentMonth = "%02d".format(now.get(Calendar.MONTH) + 1)
        val currentDay = "%02d".format(now.get(Calendar.DAY_OF_MONTH))

        val s3Path = "s3a://tweets/topics/tweet-created/" +
            s"year=$currentYear/month=$currentMonth/day=$currentDay/*"

        s3Path
    }
}

File: services/state-party-affiliation-analytic/state_party_affiliation_analytic/common/path_translator.py
from datetime import datetime, timedelta


def get_s3_path(offset: int) -> str:
    now = datetime.now()
    now = now - timedelta(days=offset)

    s3_path = f's3://tweets/topics/tweet-created/' \
              f'year={now.year}/' \
              f'month={str(now.month).zfill(2)}/day=05/*/*'
    return s3_path
